pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagentplan.go

```diff
@@ -56,3 +78,77 @@ func (h *handler) OnChangeInstallSUC(cluster *rancherv1.Cluster, status rancherv
 		mcc,
 	}, status, nil
 }
+
+// syncSystemUpgradeControllerStatus queries the managed system-upgrade-controller chart and determines if it is properly configured for a given
+// version of Kubernetes. It applies a condition onto the control-plane object to be used by the planner when handling Kubernetes upgrades.
+func (h *handler) syncSystemUpgradeControllerStatus(obj *rkev1.RKEControlPlane, status rkev1.RKEControlPlaneStatus) (rkev1.RKEControlPlaneStatus, error) {
+	sucBundle, sucBundleRetrievalError := h.bundles.Get(fleet.ClustersDefaultNamespace, name.SafeConcatName("mcc", obj.Name, "managed", "system-upgrade-controller"), metav1.GetOptions{})
+	if sucBundleRetrievalError != nil && !errors.IsNotFound(sucBundleRetrievalError) {
+		return status, sucBundleRetrievalError
+	}
+
+	rke2.SystemUpgradeControllerReady.False(&status)
+	// if we couldn't find the bundle then we know it's not ready
+	if sucBundleRetrievalError != nil {
+		return status, nil
+	}
+
+	// determine if the SUC deployment has been rolled out fully, and if there were any errors encountered
+	if sucBundle.Status.Summary.Ready != sucBundle.Status.Summary.DesiredReady {
+		if sucBundle.Status.Summary.ErrApplied != 0 && len(sucBundle.Status.Summary.NonReadyResources) > 0 {
+			nonReady := sucBundle.Status.Summary.NonReadyResources
+			rke2.SystemUpgradeControllerReady.Reason(&status, fmt.Sprintf("Error Encountered Waiting for System Upgrade Controller Deployment To Roll Out: %s", nonReady[0].Message))
```

Comment:
Does this indicate retriable errors, we probably want it to differentiate errors vs. in progress.

Reply:
Thats a good point, i'll look more into that today

Reply:
I don't believe we can determine if the error here is retriable or permanent, in either case the same fields are used. Unless we know what types of errors can be retried and check against those I'm not sure how we would differentiate